### DISCUSSION TITLE
Bioenergy demand and supply

### DISCUSSION THREAD

**Discussion by justineschafer** [magpiemodel/magpie#403] [Q&A]
Hi, I have a few questions around bioenergy demand and supply.

1. I ran the 'default' scenario, but it seems that this produces bioenergy demand numbers that don't correspond to actual primary energy demand for biomass, which is around 50-60 EJ in 2015/20. When I sum together all the Demand|Bioenergy variables (1st generation, 2nd generation, overproduction and traditional burning) in the report file, I only get to around 10-15EJ in 2015/20. What am I missing in getting these numbers to line up with actuals?
2. When feeding in non-REMIND exogenous pathways for bioenergy, what exactly should be in scope for the pathways in ‘reg.2ndgen_bioenergy_demand’ ? Is it all 2nd generation biomass, or 2nd generation biomass excluding residues?
3. Is there any more technical documentation on how the coupling between REMIND and MAgPIE works in detail? I would be particularly curious to understand how/where the bioenergy supply curves get generated (i.e. price/availability mappings) – is this something that is included in the MAgPIE code somewhere?

Thanks!

**Comment by dklein-pik** [comment]
Hi Justine,

Thank you for your interest in MAgPIE and for the questions.

1) We'll respond as soon as possible.

2) ‘reg.2ndgen_bioenergy_demand’ is the regional primary energy production (! not demand !) of second generation (ligno-cellulosic) short rotation energy crops (i.e. herbaceous and woody crops in farming systems and plantation forests grown on short rotations). Residues are excluded.

3) A very technical documentation of the REMIND-MAgPIE coupling you find [here](https://github.com/remindmodel/remind/blob/develop/tutorials/4_RunningREMINDandMAgPIE.md#what-happens-during-a-single-coupled-run). At least it contains a short description which data is exchanged in the coupling. Regarding the bioenergy supply curves: they are not part of the MAgPIE code but of the [REMIND code](https://github.com/remindmodel/remind/blob/0960ecb705b8391cefea5e89a2c7c0f641ee1372/modules/30_biomass/magpie_40/equations.gms#L41). They are simply linear regional supply curves that provide the bioenergy price as a function of bioenergy demand (for the 2nd gen biomass mentioned above). For a description of the concept, see the [vignette of the R package "remulator"](https://pik-piam.r-universe.dev/ui#view:remulator/magpie_emulator_for_remind.html) that we developed to derive the supply curves.

If you have any further questions, we will be happy to help.

**Reply by justineschafer** [reply]
Hi David, thank you for your help!

Regarding (3), I have installed the remulator package and tried running the emulator start script (as per the vignette) within MAgPIE, but am getting an error saying "Following files not found: emulator.tgz". Where could I get this file? Is this related to the 'REMIND mif file' the vignette refers to? Where could I get a file like this if REMIND is not yet fully open-source due to input data licensing (please correct me if I am wrong on this)?

If it's not possible to run emulator outside PIK, is it possible to find the standard supply curve data anywhere that is used for uncoupled runs?

**Reply by dklein-pik** [reply]
Hi Justine,

At the moment, it is indeed unfortunately not possible to calculate the curves outside PIK using PIK data. Whether I am allowed to publish the curves at the moment is unclear. For which policy scenario (SSP and RCP) would you like to have the data?

**Reply by justineschafer** [reply]
Hi David, thank you for clarifying. I am after the curves for SSP2, ideally for the broad narrative buckets of NPIs, NDCs, below-2C and 1.5C. But just NDCs would be a great start. Let me know if data sharing would depend on us having access to the same third party data, as we may already have this.

**Comment by flohump** [comment]
re 1)
What is the source for the number on actual primary energy demand for biomass of around 50-60 EJ in 2015/20?
I think the difference is explained by crop and forestry residues which are not included in the MAgPIE reporting. 
Obviously, this is something that should be improved in the future.

**Comment by justineschafer** [comment]
Hi both, 
1) I am referencing the REMIND-MAGPIE runs for the NGFS, see screenshot below. It does look like the vast majority of the Primary Energy|Biomass variable (green) is explained by the Primary Energy|Biomass|Residues variable (purple)
![image](https://user-images.githubusercontent.com/107868222/175558577-96874c0e-4c28-4d15-9885-70a633fdbe93.png)
However, the residues in MAgPIE (presumably that's the data in modules/60_bioenergy/input/f60_2ndgenBE_residue_dem) only sum up to around 5.6/6.6 EJ in 2015/20 for ssp2? I know these are not the exact same scenarios, but I would have thought total bioenergy demand coming out of MAgPIE to be in the 50-60 EJ ballpark.

Let me know if that is not clear!

Thanks very much for your help.

**Reply by flohump** [reply]
There are estimates for crop and forestry biomass residues, which enter REMIND as exognous assumptions. These estimates for crop and forestry residues have been derived largely independent from MAgPIE. Crop residues enter MAgPIE to make sure that the residues are only used once. However, foresty resides do not enter MAgPIE because a foresty sector in MAgPIE has developed only recently. Therefore, the forestry residues to not show up in the MAgPIE reporting.
We will improve the representation of residues in REMIND-MAgPIE in the future.

**Reply by justineschafer** [reply]
Hi Florian, thank you for the clarification, that is helpful - do you know where I could find that exogenous forestry residue pathway? Are there several versions/scenarios of this?
